gemrb/core/System/String.cpp

```diff
@@ -109,10 +109,12 @@ String* StringFromCString(const char* string)
 
 char* MBCStringFromString(const String& string)
 {
-	char* cStr = (char*)malloc(string.length()+1);
+	size_t allocatedBytes = string.length() * sizeof(String::value_type);
+	char *cStr = (char*)malloc(allocatedBytes);
```

Comment:
As we are about to `realloc` anyway (for shrinking in 99.99%, I would not recommend to keep `n*sizeof(String::value_type)` bytes reserved for no reason if working on let's say `n` ASCII chars), so I guess it's up to the same in the end. Would be `calloc` or `memset(_, 0, _)` vs. current approach.